frontend-wasm/src/miden_abi/transform.rs

```diff
@@ -0,0 +1,101 @@
+use miden_diagnostics::DiagnosticsHandler;
+use miden_hir::{FunctionIdent, InstBuilder, SourceSpan, Type::*, Value};
+
+use super::tx_kernel;
+use crate::module::function_builder_ext::FunctionBuilderExt;
+
+/// The strategy to use for transforming a function call
+enum TransformStrategy {
+    /// The Miden ABI function returns a length and a pointer and we only want the length
+    ListReturn,
+    /// The Miden ABI function returns on the stack and we want to return via a pointer argument
+    ReturnViaPointer,
+    /// No transformation needed
+    NoTransform,
+}
+
+/// Get the transformation strategy for a function name
+fn get_transform_strategy(function_id: &str) -> TransformStrategy {
+    match function_id {
+        tx_kernel::NOTE_GET_INPUTS => TransformStrategy::ListReturn,
+        tx_kernel::ACCOUNT_ADD_ASSET => TransformStrategy::ReturnViaPointer,
+        tx_kernel::ACCOUNT_GET_ID => TransformStrategy::NoTransform,
+        _ => panic!("No transform strategy found for function {}", function_id),
+    }
+}
+
+/// Transform a function call based on the transformation strategy
+pub fn transform_miden_abi_call(
+    func_id: FunctionIdent,
+    args: &[Value],
+    builder: &mut FunctionBuilderExt,
+    span: SourceSpan,
+    diagnostics: &DiagnosticsHandler,
+) -> Vec<Value> {
+    use TransformStrategy::*;
+    match get_transform_strategy(func_id.function.as_symbol().as_str()) {
+        ListReturn => list_return(func_id, args, builder, span, diagnostics),
+        ReturnViaPointer => return_via_pointer(func_id, args, builder, span, diagnostics),
+        NoTransform => no_transform(func_id, args, builder, span, diagnostics),
+    }
+}
+
+/// No transformation needed
+#[inline(always)]
+pub fn no_transform(
```

Comment:
Just as a heads up, we should make sure that any function type which would be handled as `no_transform`, uses `Abi::Canonical` so that it goes through the normal lowering process. For the other Abi variants, we either want to transform them such that they are `Canonical`, or that they require no further transformation aside from direct translation to Miden Assembly.

To be clear, haven't checked to see if that's already the case anyway, but wanted to raise it just in case that conflicts with what your operating assumptions are.

Reply:
Good point! Actually, since all the Miden ABI transformation happens in the frontend, I'll use `Abi::Canonical` for all the function type definitions because all of them should have this ABI after the processing regardless whether an actual transformation was conducted. This means that `Abi::Miden` is not used, so it can be deleted. Keep in mind that `Abi` is not included in [`Signature`](https://github.com/0xPolygonMiden/compiler/blob/cfabce971c3357f2feb3d559406be3ea97890468/hir/src/function.rs#L160-L175). Besides that, I removed `(abi XXXXX)` in `ComponentImport` textual form and printed it as part of `FunctionType` in the textual form. 
All the aforementioned changes are made in the #156 due to their invasive nature to the two subsequent PRs that makes rebasing a PITA. See commit https://github.com/0xPolygonMiden/compiler/pull/155/commits/82870e39cb379f04e3a5a5d370e9734377b633f7